text/0000-fabric-private-chaincode.md

```diff
@@ -0,0 +1,284 @@
+---
+layout: default
+title: Fabric Private Chaincode
+nav_order: 3
+---
+ 
+- Feature Name: Fabric Private Chaincode (FPC)
+- Start Date: 2020-01-28)
+- RFC PR: (leave this empty)
+- Fabric Component: Core (fill me in with underlined fabric component, core, orderer/consensus and etc.)
+- Fabric Issue: (leave this empty)
+ 
+# Summary
+[summary]: #summary
```

Comment:
Fundamentally, in a blockchain network, each organization could run a peer that is implemented in a different language as compared to others. This is true in the Ethereum world. The hard requirement is that majority of the node should follow the predefined rules. I believe that the same is applicable for Fabric too. Now, we are adding a constraint that all peers must use hardware with a specific feature. I am not sure about the feasibility. For example, I might host my peers on-prem and might have a very strong security team to keep the attackers away from my peers. Whereas, some other organization in the same network might be using a cloud vendor to host the peers which may require a TEE. I am not sure whether it is good to enforce every channel member to use specific hardware.

Reply:
I completely agree. Eventually, FPC should also follow these principles and support heterogeneous TEE platforms and let the users decide on endorsement policies that reflect their needs. With SGX we are just starting to explore the TEE capabilities in the context of blockchain.